tests/linearizability/linearizability_test.go

```diff
@@ -133,6 +136,62 @@ func testLinearizability(ctx context.Context, t *testing.T, clus *e2e.EtcdProces
 	return operations, events
 }
 
+func patchOperationHistory(operations []porcupine.Operation, watchOps []watchEvent) []porcupine.Operation {
+	newOperations := make([]porcupine.Operation, 0, len(operations))
+	persisted := map[EtcdOperation]watchEvent{}
+	maxTime := watchOps[0].Time
+	for _, op := range watchOps {
+		persisted[op.Op] = op
+		if op.Time.After(maxTime) {
+			maxTime = op.Time
+		}
+	}
+
+	for _, op := range operations {
+		request := op.Input.(EtcdRequest)
+		resp := op.Output.(EtcdResponse)
+		if resp.Err == nil || op.Return > maxTime.UnixNano() {
+			// Leave successful and not observed requests as they are
+			newOperations = append(newOperations, op)
+			continue
+		}
+		hasPut := false
+		wasPersisted := false
+		hasDelete := false
+		for _, op1 := range request.Ops {
+			switch op1.Type {
+			case Delete:
+				hasDelete = true
+			case Put:
+				hasPut = true
+				op2, ok := persisted[op1]
```

Comment:
watchedOp, := ...

Reply:
Thanks for pointing this out. Forgot about this :P